### PR TITLE
Add create/deploy release Github workflow

### PR DESCRIPTION
**Pull request by jordanguedj**
A Github workflow to trigger our Gitlab CI/CD pipelines.
Workflows are triggered when a push is made to the `main` branch.
It's only enabled for development environment (for now).

### CODE REVIEW

File: .github/workflows/create-deploy-release.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+on:
+  push:
+    branches:
+      - main
+jobs:
+  trigger-gitlab-pipeline:
+    runs-on: ubuntu-latest
+    env:
+      # static variables and secrets to be defined
+      APP_NAME: ${{ vars.APP_NAME }}
```
Comment (maudetes) [inline .github/workflows/create-deploy-release.yml:12]: These vars should be configured in the repo settings?

File: .github/workflows/create-deploy-release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          ssh-keyscan -t rsa gitlab.com >> ~/.ssh/known_hosts
+
+          # configure git
+          git config --global user.email "root@data.gouv.fr"
+          git config --global user.name "datagouv"
+
+          # set ENV based on branch
+          if [ "$GITHUB_REF" = "refs/heads/main" ]; then
+            echo "ENV=dev" >> $GITHUB_ENV
+          fi
```
Comment (maudetes) [inline .github/workflows/create-deploy-release.yml:35]: I don't think we need this condition in our case?

File: .github/workflows/create-deploy-release.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - name: clone scaffold repository
+        shell: bash
+        run: |
+          git clone --quiet --depth 1 $SCAFFOLD_REPO_SSH_URL $SCAFFOLD_DIR
+      - name: trigger Gitlab CI/CD pipeline
+        shell: bash
+        run: |
+          RELEASE_VERSION="${GITHUB_SHA:0:8}"
```
Comment (maudetes) [inline .github/workflows/create-deploy-release.yml:43]: :+1:

**Review by maudetes — APPROVED** [review]

File: .github/workflows/create-deploy-release.yml (hunk 1 above)
Comment (jordanguedj) [reply .github/workflows/create-deploy-release.yml:12]: Yes, and it's already done for this repository.

**Pull request merged**